W-12241095-apikit-graphql-api-implementation-CP (#112)
mulesoft/docs-apikit: modules/ROOT/pages/apikit-graphql-api-implementation.adoc

= Querying a GraphQL API implementation
ifndef::env-site,env-github[]
include::_attributes.adoc[]
endif::[]

> This section showcases the implementation example of a GraphQL service to query data using APIkit for GraphQL.

Reviewer: ```suggestion
The following examples showcase the implementation of a GraphQL service to query data using APIkit for GraphQL.
```

Author: Thanks!

> To run the set of examples described below, download and run the *apikit-graphql-example* application, which loads a sample dataset in memory for testing purposes. Get more information about https://graphql.org/learn/serving-over-http/[the format of the queries sent through an HTTP POST request^].

Reviewer: ```suggestion
Before running these examples, download and run the *apikit-graphql-example* application, which loads a sample dataset in memory for testing purposes. Get more information about https://graphql.org/learn/serving-over-http/[the format of the queries sent through an HTTP POST request^].
```

Author: Thanks!

               "body": "Sed ante. Vivamus tortor. Duis mattis egestas metus.\n\nAenean fermentum. Donec ut mauris eget massa tempor convallis. Nulla neque libero, convallis eget, eleifend luctus, ultricies eu, nibh.\n\nQuisque id justo sit amet sapien dignissim vestibulum. Vestibulum ante ipsum primis in faucibus orci luctus et ultrices posuere cubilia Curae; Nulla dapibus dolor vel est. Donec odio justo, sollicitudin ut, suscipit a, feugiat et, eros."
           },
           ...
           {
            "id": "87",
               "title": "nam tristique",
               "body": "Proin eu mi. Nulla ac enim. In tempor, turpis nec euismod scelerisque, quam turpis adipiscing lorem, vitae mattis nibh ligula nec sem.\n\nDuis aliquam convallis nunc. Proin at turpis a pede posuere nonummy. Integer non velit.\n\nDonec diam neque, vestibulum eget, vulputate ut, ultrices vel, augue. Vestibulum ante ipsum primis in faucibus orci luctus et ultrices posuere cubilia Curae; Donec pharetra, magna vestibulum aliquet ultrices, erat tortor sollicitudin mi, sit amet lobortis sapien sapien non mi. Integer ac neque."
           }
       ]
   }
}

=== Retrieve All Users with Data Loader

> This query requests the data of all the users, including their address information. In terms of implementation, this is using a Data Loader to fetch the addresses of each user in only one request.

Reviewer: ```suggestion
This query requests the data of all users, including their address information. In terms of implementation, this example uses a Data Loader to fetch the addresses of each user in only one request.
```

Author: Thanks!

--header 'Content-Type: application/json' \
--data-raw '{"query":"mutation ReplyToComment($commentId:ID!,$userId:ID!,$body:String!){replyToComment(commentId:$commentId,userId:$userId,body:$body){id body post{id title comments{id body replies{id user{username}body}}}}}","variables":{"commentId":201,"userId":42,"body":"Thank you!"}}'

==== Get a Response

{
   "data": {
       "replyToComment": {
           "id": "202",
           "body": "Thank you!",
           "post": {
               "id": "101",
               "title": "GraphQL by example",
               "comments": [
                   {
                       "id": "201",
                       "body": "Nice post! Thanks for sharing! It's just what I was looking for",
                       "replies": [
                           {
                               "id": "202",
                               "user": {
                                   "username": "tblackmore15"
                               },
                               "body": "Thank you!"
                           }
                       ]
                   }
               ]
           }
       }
   }

> }

Reviewer: Please use code formatting for each code sample in the article.

Author: Thanks for mentioning it!

== See Also

> * xref:doc.adoc[Some Relevant Doc Article]

Reviewer 2: Remove this

Author: I'm still thinking about what link could be included, and if it should be included. So I'll leave this comment as a reminder ;)
Thanks!
